### PR TITLE
`Development`: Fix a date conversion error for quizzes

### PR DESCRIPTION
**Pull request by manuelmanso**
#### General
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

Fixes the Cypress tests that were failing for quiz exercises.

### Description

Small bugfix in complaint service that flooded console with errors and caused component not to render.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student

1. Log in to Artemis
2. Navigate to Course Administration

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Test Coverage
| Class/File | Branch | Line |
|------------|-------:|-----:|

### CODE REVIEW

File: src/main/webapp/app/complaints/complaint.service.ts [hunk 1]
```diff
@@ -249,7 +249,7 @@ export class ComplaintService implements IComplaintService {
             return undefined;
         }
 
-        return complaintStartDate!.add(course.maxComplaintTimeDays!, 'days');
+        return dayjs(complaintStartDate).add(course.maxComplaintTimeDays!, 'days');
```
Comment (krusche) [inline src/main/webapp/app/complaints/complaint.service.ts:252]: that's a good workaround and bugfix (and we should keep it), but we should make sure that all dates are already converted directly after they are downloaded from the server in the corresponding method that executes the http call.

**Review by Strohgelaender — APPROVED** [review]
Code looks good and the test were sucessfull as well.

**Comment by krusche** [comment]
I will merge this now as important bugfix, additional improvements can be made in a follow-up PR

**Review by TheZoker — COMMENTED** [review]
Can confirm this works!

**Pull request merged**